third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1074,9 +1074,52 @@ struct FDivOpConversion
                                    ConversionPatternRewriter &rewriter,
                                    Type elemTy, MultipleOperandsRange operands,
                                    Location loc) const {
+    // For non-F32 input, it's lowered to LLVM::FDivOp, which is a
+    // IEEE-compliant DIV operation.
+    if (elemTy.getIntOrFloatBitWidth() != 32)
+      return {rewriter.create<LLVM::FDivOp>(loc, elemTy, operands[0][0],
+                                            operands[0][1])};
 
-    return {rewriter.create<LLVM::FDivOp>(loc, elemTy, operands[0][0],
-                                          operands[0][1])};
+    // The algorithm comes from
+    // https://github.com/llvm/llvm-project/blob/bda7aadfcd0596675c8b4b381f178f3d8f854f20/llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp#L4980-L5065
```

Comment:
Nit: you can shorten the URL a bit by just having the first 8 or 12 chars in the commit sha.